Fix FlexContariner guides (#1272)
Samsung/tizen-docs: docs/application/dotnet/guides/nui/flexcontainer.md

![Flex container terminology](media/flex-container.jpg)

> [!NOTE]
> Use FlexLayout instead of FlexContainer class. FlexContainer class is deprecated since Tizen 6.0 and will be supported for the subsequent two releases only.

<a name="layoutexample"></a>
## Creating a Flexbox layout

The following example shows how to create a gallery-like flexbox layout using the [Tizen.NUI.BaseComponents.FlexContainer](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.FlexContainer.html) object.

**Figure: Flex container example**

![Flex container example](media/flexbox-demo.jpg)

1.  Add the following namespaces:

    ```csharp
    using Tizen.NUI;
    using Tizen.NUI.BaseComponents;
    using Tizen.NUI.UIComponents;
    ```

> 2.  Create a flex container as the whole view by setting the resize policy to fill its parent, the window:

Reviewer: @pkalota  here I have a confusion, can you please help me understand:
- its parent ? 
what is it referring to?
can we write the subject name instead of **its**
Is **its** referring to **the window**?
Could you please rephrase this?
2.  Create a flex container as the whole view by setting the resize policy to fill its parent, the window:

Author: Yes, its parent refers to window. I hope the following change gives more clarity.
`2.  Create a flex container as the whole view by setting the resize policy to fill the window:`

![Flex container terminology](media/flex-container.jpg)

> [!NOTE]

> > FlexContainer is deprecated in API8 and will be removed in API10. Please use FlexLayout instead!

Reviewer: @pkalota Just I have a doubt.
Here FlexContainer refers to?

a class ?? or 

As per definition: A flex container is a NUI container control that contains items. 

can we say: 

> [!NOTE]
> FlexContainer is deprecated in API level 8 and will be removed in API level 10. Please use FlexLayout instead!

> > FlexContainer is deprecated in API8 and will be removed in API10. Please use FlexLayout instead!

Author: I wanted to share information that FlexContainer Class is deprecated which is displayed here https://samsung.github.io/TizenFX/API8/api/Tizen.NUI.BaseComponents.FlexContainer.html

> > FlexContainer class is deprecated in API level 8 and will be removed in API level 10. Please use FlexLayout instead!

Reviewer: ```suggestion
> Use FlexLayout instead of FlexContainer class. FlexLayout class is deprecated since Tizen 6.0 and will be supported for subsequent two releases only .
```

Just aligning the comeent with @mijin85cho comment https://github.com/Samsung/tizen-docs/pull/1272/files#r499959363

![Flex container example](media/flexbox-demo.jpg)

1.  Add namespace using:

    ```cs
    using Tizen.NUI;
    using Tizen.NUI.BaseComponents;
    using Tizen.NUI.UIComponents;
    ```

> 2.  Create a flex container as the whole view and set its resize policy to fill its parent (the window):

Reviewer: Just a suggestion, can we write like this?
If you dont like, kindly ignore.
```suggestion
2.  Create a flex container as the whole view by setting the resize policy to fill its parent (the window):

>     ```cs

Reviewer 2: Global.  Would you please refer https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#supported-languages? It seems we should use "csharp", not "cs".

>     Window.Instance.Add(flexContainer);

Author: In our examples `Window.Instance.Add(flexContainer)` and `Window.Instance.GetDefaultLayer().Add(flexContainer)` makes no differenece. Therefore I will change all `Window.Instance.GetDefaultLayer().Add(flexContainer)`  to  `Window.Instance.Add(flexContainer)` to unify examples.

    ```

3.  To make the toolbar and main content display vertically, set the flex direction of the main container to **column**:

    ```cs
    /// Display toolbar and content vertically
    flexContainer.FlexDirection = FlexContainer.FlexDirectionType.Column;
    ```

> 4.  Create a new flex container as the toolbar and add it to the main container. Because the flex direction in the main container is **column**, the toolbar is placed at the top.

Reviewer: How about a slight change in sentence structure to make it more clear. Please check for technical accuracy too.
```suggestion
4.  Create a new flex container as the toolbar and add it to the main container. Since the flex direction in the main container is set to **column**, the toolbar is placed at the top.

> 4.  Create a new flex container as the toolbar and add it to the main container. Since the flex direction in the main container is set to **column**, the toolbar is placed at the top.

Reviewer: Kindly add : instead of . if the sentence is immediately followed by code. 
```suggestion
4.  Create a new flex container as the toolbar and add it to the main container. Since the flex direction in the main container is set to **column**, the toolbar is placed at the top:

    flexContainer.Add(toolBar);
    ```

> 5.  To make the buttons and title display horizontally, but vertically aligned to the center of the toolbar with equal spaces between them, set the toolbar flex direction to **row**, its `AlignItems` property to **center** and `JustifyContent` property to **spaceBetween**:

Reviewer: How about this slight modification, request you check for technical accuracy: 
```suggestion
5.  To make the buttons and title display horizontally, with vertical alignment to the center of the toolbar, and  having equal space between them, set the toolbar flex direction to **row**, its `AlignItems` property to **center** and `JustifyContent` property to **spaceBetween**:

> 5.  To make the buttons and title display horizontally, with vertical alignment to the center of the toolbar and having equal space between them, set the toolbar flex direction to **row**, its `AlignItems` property to **center** and `JustifyContent` property to **spaceBetween**:

Reviewer: @pkalota Just broke the sentence for better comprehension. 
```suggestion
5.  To make the buttons and title display horizontally, with vertical alignment to the center of the toolbar, and having equal space between them. Set the toolbar flex direction to **row**, its `AlignItems` property to **center**, and `JustifyContent` property to **spaceBetween**:

Reviewer 2: @nawab-samsung Would you please check this suggestion again? It is breaking sentence into uncompleted sentences.
The first sentence becomes weird with this update.

Reviewer: @mijin85cho  sure

Author: @nawab-samsung maybe it is better to leave comma before `set`? By setting dot there you are cutting first sentence, which started with `To make the buttons and title display horizontally...`, so we should provide explanation how to make it in the same sentence - `, set the toolbar flex direction to **row**...`. In case of your suggestion sentence seems to be uncompleted as @mijin85cho mentioned before.

You are proposing 2 commas before `and` as well. As far as I know, let me know if I'm wrong, in this case both of solutions are correct, but in my mind it looks better without commas. What about leaving it without any change?

Generally it is good idea to split this long sentence. What do you think @newab-samsung and @mijin85cho about following description of this subpoint?

5. To change display of toolbar set the `FlexDirection` to **row**, its `AlignItems` property to **center** and `JustifyContent` property to **spaceBetween**. In result buttons and title will be displayed horizontally, with vertical alignment to the center of the toolbar and having equal space between

Reviewer 2: I think changing the sentence in this way is much better than just cutting the sentence. Adding to that, how about some modification like below?

Change the display of toolbar by setting the `FlexDirection` to **row**, `AlignItems` property to **center**, and **JustifyContent** property to **spaceBetween**. It displays buttons and title horizontally, with vertical alignment to the center of the toolbar and having equal space between them.

Reviewer: thanks @pkalota for making me understand this. 

Yes @mijin85cho your suggestion w.r.t to current rephrase is good 
```suggestion
Change the display of toolbar by setting the `FlexDirection` to **row**, `AlignItems` property to **center**, and `JustifyContent` property to **spaceBetween**. It displays buttons and title horizontally with vertical alignment to the center of the toolbar and having equal space between them.

Reviewer 2: Thanks for checking @nawab-samsung. I also think your latest suggestion in your comment is acceptable. 
@pkalota Would you please check and give us your opinion?

Author: I think that whole sentence looks better now and it is easier to understand this subpoint. I will apply it in next push.

    toolBar.JustifyContent = FlexContainer.Justification.JustifySpaceBetween;
    ```

> 6.  Use `Flex` property of the toolbar to specify occupied area in main container by it. The main container's height is shared between toolbar and content (added in step 7). Since main container displays items vertically and toolbar `Flex` property is set to 0.1, toolbar will occupy 10% of verical and 100% of horizontal available space. Remainded area is intended for content:

Reviewer: Please check for technical accuracy.

```suggestion
6.  Use `Flex` property of the toolbar to specifes the area to be occupied within the main container. The height main container is shared between toolbar and content as specified in next step. Since the main container displays items vertically and the toolbar `Flex` property is set to 0.1, the toolbar will occupy 10% of vertical and 100% of horizontal available space. The remaining area is intended for the content:

Reviewer 2: Before @pkalota's check, would you please check the "to specifes"? "to specify" seems correct here.

Reviewer: Sure, thanks

> 

Author: @nawab-samsung I changed description of this subpoint a little. Please share your opinion wheter usage of `Flex` property is clearly explained here.

Reviewer: @pkalota sure will check

> 6.  Use `Flex` property of the toolbar to specify the area to be occupied within the main container. The height main container is shared between toolbar and content as specified in next step. Since the main container displays items vertically and the toolbar `Flex` property is set to 0.1, the toolbar will occupy 10% of vertical and 100% of horizontal available space. The remaining area is intended for the content:

Reviewer: Slight modification needed ( you had missed **of**) from previous comment 

```suggestion
6.  Use `Flex` property of the toolbar to specify the area to be occupied within the main container. The height of the main container is shared between the toolbar and the content as specified in the next step. Since the main container displays items vertically and the toolbar `Flex` property is set to 0.1, the toolbar will occupy 10% of vertical and 100% of horizontal available space. The remaining area is intended for the content:

Reviewer 2: One opinion to "as specified in the next step"

The original "added in step 7" was decorating "content". So the meaning was "the content which is added in step 7"
However it is changed to "as specified in the next step", so it becomes to decorate "The height of the main container is shared between the toolbar and the content".

The meaning is changed, and even more, in step 7, "creating a content"is described rather than "sharing the height of the main container between the toolbar and the content". "Occupy 90 percent of available space on the cross axis" is kind of description of the content property, instead of focusing "sharing the height".

So I think we'd batter
1. to remove "as specified in the next step", as it is wrong and redundant, or
2. change  "content as specified in the next step" to other, for example like "content that will be created", if you want to keep.

In my thought removing is better, but I'd like to get your opinion. Would you please let me know your thought?

Reviewer: Sure, @mijin85cho Although, literally both mean the same though not completely, 

added in step 7 

as specified in the next step

 The height of the main container is shared between the toolbar and the content as added in (step 7). 
 The height of the main container is shared between the toolbar and the content as specified in the next step.

But yes your suggestion is good, but may not be relevant, as step 6 is talking about specifically the height/position/ location, thats where step 7 gets reffered to. 

@pkalota could you please provide your valuable opinion on this.

Author: Maybe it is better to firstly create toolbar and content and after that focus on flex property? It is more clear to put `Flex` properties of both items next to them. It shows connection between `Flex` properties of these items and it will be easier for user to understand this dependency. @mijin85cho and @nawab-samsung please take a look at below change and let me know what are you thinking about that.

6.  To display the image, create the content area by adding a third flex container at the bottom of the main container. Show the image in the center of the content area container by centering the item horizontally and vertically inside the area:

    ```csharp
    /// Create the content area
    FlexContainer content = new FlexContainer();
    content.ParentOrigin = ParentOrigin.TopLeft;
    content.PivotPoint = PivotPoint.TopLeft;
    /// Place items horizontally
    content.FlexDirection = FlexContainer.FlexDirectionType.Row;
    /// Align items horizontally center
    content.JustifyContent = FlexContainer.Justification.JustifyCenter;
    /// Align items vertically center
    content.AlignItems = FlexContainer.Alignment.AlignCenter;
    /// Add the content area to the main container
    flexContainer.Add(content);
    ```

7.  Use `Flex` property of the `FlexContainer` class to specify the area to be occupied within the main container. Since the main container displays items vertically its height is shared between the toolbar and the content. Setting `toolbar.Flex` to 0.1 and `content.Flex` to 0.9 means that `toolbar` will occupy 10% of vertical space and `content` will occupy 90% of vertical space of the main container. Width of items will be eqaul to width of main container.

    ```csharp
    /// Occupy 10 percent of available space on the cross axis
    toolBar.Flex = 0.1f;
    /// Occupy 90 percent of available space on the cross axis
    content.Flex = 0.9f;
    ``` 

If you prefer previous version, I think 2nd @mijin85cho's option 

>change "content as specified in the next step" to other, for example like "content that will be created", if you want to keep."

is the best one here. Thanks that we are avoiding user's question "What is this content?".

    ```csharp
    /// Occupy 10 percent of available space on the cross axis
    toolBar.Flex = 0.1f;
    ```

> 7.  To display the image, create the content area by adding a third flex container at the bottom of the main container. Show the image in the center of the content area container by centering the item horizontally and vertically inside the area:

Reviewer: ```suggestion 
7.  To display the image, create the content area by adding a third flex container at the bottom of the main container. You can show the image in the center of the content area container by centering the item horizontally and vertically inside the area:

Reviewer 2: @nawab-samsung How about removing redundancies? I think your "You can" can be omitted here.

Reviewer: @mijin85cho  to better the sentence structure I used **you can**. 
Directly telling show doesnt go well with the sentence flow. 
If you do not agree, we can go with your suggestion . Kindly Let me know  your opinion,

    toolBar.JustifyContent = FlexContainer.Justification.JustifySpaceBetween;
    ```

> 6.  Use the toolbar's `Flex` property to make the toolbar and the main content share the height of the main container, so that the toolbar occupies 10 percent of the vertical space and the content occupies the remainder:

Reviewer: I couldn't get this clearly but tried to comprehend what is being conveyed, kindly check if this is as per your intention, or else can you slightly elaborate? 

```suggestion
6.  Use the `Flex` property of the toolbar to make the toolbar. The main content shares the height of the main container so that the toolbar occupies 10 percent of the vertical space and the content occupies the remainder space:

    ```cs
    /// Occupy 10 percent of available space on the cross axis
    toolBar.Flex = 0.1f;
    ```

7.  To display the image, create the content area by adding a third flex container at the bottom of the main container.

>     Show the image in the center of the content area container by centering the item horizontally and vertically inside the area.

Reviewer: ```suggestion
    Show the image in the center of the content area container by centering the item horizontally and vertically inside the area:

    content.JustifyContent = FlexContainer.Justification.JustifyCenter;
    /// Align items vertically center
    content.AlignItems = FlexContainer.Alignment.AlignCenter;
    /// Add the content area to the main container
    flexContainer.Add(content);
    ```

> 7.  Use `Flex` property of the `FlexContainer` class to specify the area to be occupied within the main container. Since the main container displays items vertically its height is shared between the toolbar and the content. Setting `toolbar.Flex` to 0.1 and `content.Flex` to 0.9 means that `toolbar` will occupy 10% of vertical space and `content` will occupy 90% of vertical space of the main container. Width of items will be eqaul to width of main container.

Reviewer: ```suggestion
7.  Use `Flex` property of the `FlexContainer` class to specify the area to be occupied within the main container. Since, the main container displays items vertically, its height is shared between the toolbar and the content. Setting `toolbar.Flex` to 0.1 and `content.Flex` to 0.9 means that `toolbar` will occupy 10% of vertical space and `content` will occupy 90% of vertical space of the main container. Thus, the width of the items equals the width of the main container:

Reviewer 2: I have two questions : 

> Since,

I'm not sure with the utility of adding comma after "Since". How about not adding the comma after "Since"?

>Thus, the width 

I'm not sure with the "Thus" here. The previous sentence is talking about **vertical height**, so why "Thus" is added to **width issue**?

Reviewer: @mijin85cho both are correct, I took your suggestion as it looks fine. 
Why I had added (,) was to give a slight pause. 
I had added "thus" to give a notion of inference.

> 7.  Use `Flex` property of the `FlexContainer` class to specify the area to be occupied within the main container. Since the main container displays items vertically its height is shared between the toolbar and the content. Setting `toolbar.Flex` to 0.1 and `content.Flex` to 0.9 means that `toolbar` will occupy 10% of vertical space and `content` will occupy 90% of vertical space of the main container. The width of the items will be eqaul to the width of the main container:

Reviewer: @pkalota slight correction in spelling and adding a , 
```suggestion
7.  Use `Flex` property of the `FlexContainer` class to specify the area to be occupied within the main container. Since the main container displays items vertically, its height is shared between the toolbar and the content. Setting `toolbar.Flex` to 0.1 and `content.Flex` to 0.9 means that `toolbar` will occupy 10% of vertical space and `content` will occupy 90% of vertical space of the main container. The width of the items will be equal to the width of the main container:

    ```csharp
    /// Occupy 10 percent of available space on the cross axis
    toolBar.Flex = 0.1f;
    /// Occupy 90 percent of available space on the cross axis
    content.Flex = 0.9f;
    ``` 

8.  Add items to the toolbar.

    The toolbar has 1 button on the left, 1 button on the right, and a title in the center. To keep the title in the center, make the title flexible so that it automatically takes all remaining horizontal space. To make the layout look nicer, also add some space around the items.

>     Creating the toolbar with a flex container allows you to keep the toolbar layout consistent, even as it expands or reduces in size based on screen size and orientation.

Reviewer: Global: 
If the sentences are/is immediately followed by code or image, request you to ensure to add : instead of . at the end of the sentences.
```suggestion to 
    Creating the toolbar with a flex container allows you to keep the toolbar layout consistent, even as it expands or reduces in size based on screen size and orientation:

    nextButton.Label = labelMap2;
    ```

9.  Add the image to the content area:

    ```csharp
    /// Add an image to the center of the content area
    ImageView imageView = new ImageView(DirectoryInfo.Resource + "image.png");
    imageView.ParentOrigin = ParentOrigin.TopLeft;
    imageView.PivotPoint = PivotPoint.TopLeft;
    content.Add(imageView);
    ```

Full example can be found here: ...

> 

Author: I'm waiting for merging of this example https://github.com/Samsung/Tizen-CSharp-Samples/pull/176, after that I will add missing link here. This is just a copy of above snippets. It will be more conviniece for user to download it from here and run.

Reviewer: @mijin85cho yes, thats the form we use to let readers know that there is more to it. I actually agree with your opinion. 
As @mijin85cho suggested, we can use it in the following form:
```suggestion 
For more information, see [the full example](download link).

> For more information, see [the full example](download link)

Reviewer 2: Now this PR is approved as the comments are applied. However I'm waiting this sample PR is merged and the line 167 is updated with actual sample link. 
As now invalid link is used for the hyperlink, the hyperlink is not working.

![image](https://user-images.githubusercontent.com/9958374/96946312-f8a79f80-151a-11eb-8122-4ec69fc55c40.png)

> For more information, see [the full example](download link)

Author: Since PR is approved and it is waiting to be merged only cause of lack of sample link, I will delete it right now and add it later in another PR when sample will be available.

    nextButton.Label = labelMap2;
    ```

9.  Add the image to the content area:

    ```csharp
    /// Add an image to the center of the content area
    ImageView imageView = new ImageView(DirectoryInfo.Resource + "image.png");
    imageView.ParentOrigin = ParentOrigin.TopLeft;
    imageView.PivotPoint = PivotPoint.TopLeft;
    content.Add(imageView);
    ```

For more information, see [the full example](download link)

<a name="properties"></a>

> ## Modifying FlexContainer Properties

Reviewer: Global: 
Can we please use sentence case for headings?

Author: In other guides headings looks like this one. Are you sure that we should apply this?

    **Usage:**

    ```csharp
    FlexContainer flexContainer = new FlexContainer();
    flexContainer.ContentDirection = FlexContainer.ContentDirectionType.RTL;
    ```

-   `FlexDirection`

>     This property specifies the main axis direction along which flex items are placed. The default property is `Column`.

Reviewer: ```suggestion
    This property specifies the main axis direction along which flex items are placed. The default property is `Column`:

>     This property specifies the main axis direction along which flex items are placed. The default property is `Column`.

Reviewer: Since the sentence is immediately followed by an image, so we need : instead of .
https://github.com/Samsung/tizen-docs/pull/1272/commits/c850e531bc65eff63e37543f732ae4c2dc1f4744#r505164016

Author: So I will apply it in all sentences, which is immediately followed by an image

Reviewer: @pkalota Yes, if a sentence(s) is followed image or code then **:** is used

    **Usage:**

    ```csharp
    FlexContainer flexContainer = new FlexContainer();
    flexContainer.JustifyContent = FlexContainer.Justification.JustifySpaceBetween;
    ```

-   `AlignItems`

>     This property specifies the alignment for flex items when they do not use all the available space on the cross axis:

Reviewer: ```suggestion     
    This property specifies the alignment for flex items when all the available space on the cross axis is not used:

Reviewer: @pkalota can you please incorporate this too ?

Author: Sorry, I missed that. Done.

    **Usage:**

    ```csharp
    FlexContainer flexContainer = new FlexContainer();
    flexContainer.AlignItems = FlexContainer.Alignment.AlignFlexStart;
    ```

-   `AlignContent`

>     This property specifies the alignment for flex lines when they do not use all the available space on the cross axis. It is only valid when there are multiple lines:

Reviewer: ```suggestion     
    This property specifies the alignment for flex items when all the available space on the cross axis is not used. It is only valid when there are multiple lines:

    **Usage:**

    ```csharp
    FlexContainer flexContainer = new FlexContainer();
    flexContainer.AlignContent = FlexContainer.Alignment.AlignFlexEnd;
    ```

> <a name="childproperty"></a>

Reviewer: </a name="childproperty"/></a> 

Is this needed?

    **Figure: Flex**

    ![Flex](media/flex.png)

    To create a layout with the proportions shown in the above figure:

    ```csharp
    /// Create the flex container
    FlexContainer flexContainer = new FlexContainer();

    /// Resize flex container to window size
    flexContainer.WidthResizePolicy = ResizePolicyType.FillToParent;

>     flexContainer.HeightResizePolicy = ResizePolicyType.FillToParent;

Reviewer 2: With the updated image, I guess this example's FlexDirection is FlexDirectionType.Column.

Author: Yes, you are right. `FlexDirectionType.Column` is used by default in `FlexContainer `that's why I'm not assigning this property here. @mijin85cho please let me know if I should add it anyway.

Reviewer 2: @pkalota No, you don't need to add it here. I'd like to suggest to add the description that `Column` is the default property at line 213 or line 203.

For example in case of line 203,

This property specifies the main axis direction along which flex items are placed. The default property is `Column`.

How about this update?

    To create a layout with the proportions shown in the above figure:

    ```cs
    /// Create the flex container
    FlexContainer flexContainer = new FlexContainer();

    /// Resize flex container to window size
    flexContainer.WidthResizePolicy = ResizePolicyType.FillToParent;
    flexContainer.HeightResizePolicy = ResizePolicyType.FillToParent;

    /// Add the container to the window
    Window.Instance.GetDefaultLayer().Add(flexContainer);

    /// Create flex items and set their proportions
    View item1 = new View();
    item1.BackgroundColor = Color.Red;
    item1.Flex = 1.0f;
    flexContainer.Add(item1);

    View item2 = new View();
    item2.BackgroundColor = Color.Green;
    item2.Flex = 3.0f;
    flexContainer.Add(item2);

    View item3 = new View();
    item3.BackgroundColor = Color.Blue;
    item3.Flex = 1.0f;
    flexContainer.Add(item3);

    View item4 = new View();
    item4.BackgroundColor = Color.Yellow;
    item4.Flex = 2.0f;
    flexContainer.Add(item4);

    View item5 = new View();

>     item5.BackgroundColor = Color.White;

Reviewer 2: With this update, the background color of the items varies. However in the sample images, the background color is kept. I have a question about this. Do you have any plan to update the images also to match to the sample codes?

Author: I have treated it as a general view. But if we want to show directly this example, I can insert here screenshot from this example.

    To create the item alignment shown in the above figure:

    ```cs
    /// Create the flex container
    FlexContainer flexContainer = new FlexContainer();

    /// Set the default alignment along the cross axis to the beginning of the container
    flexContainer.AlignItems = FlexContainer.Alignment.AlignFlexStart;
    flexContainer.WidthResizePolicy = ResizePolicyType.FillToParent;
    flexContainer.HeightResizePolicy = ResizePolicyType.FillToParent;
    flexContainer.FlexDirection = FlexContainer.FlexDirectionType.Row;
    Window.Instance.GetDefaultLayer().Add(flexContainer);

    /// Create flex items and add them to the flex container
    View item1 = new View();

    /// Align item1 to the center of the container
    item1.AlignSelf = (int)FlexContainer.Alignment.AlignCenter;
    item1.Size2D = new Size2D(200, 200);
    item1.BackgroundColor = Color.Red;
    flexContainer.Add(item1);

    View item2 = new View();

    /// Align item2 to the beginning of the container based on the default alignment
    item2.Size2D = new Size2D(200, 200);
    item2.BackgroundColor = Color.Green;

>     flexContainer.Add(item2);

Reviewer 2: Question. According to the "above figure", the second item should has "flexStart" property. Don't we need to set "flexStart" to item2?

Author: As it is said in comment: 
> Align item2 to the beginning of the container based on the default alignment

Lack of `AlignSelf` is intended to show alignment inheritance

    View item3 = new View();

    /// Align item3 to the bottom of the container
    item3.AlignSelf = (int)FlexContainer.Alignment.AlignFlexEnd;
    item3.Size2D = new Size2D(200, 200);
    item3.BackgroundColor = Color.Blue;
    flexContainer.Add(item3); 

>     ```

Reviewer 2: Question. According to "above figure", we have 4 items. The item4 is removed with this update and not added back. Don't we need to add item4?

Author: To comply with above figure we can add 4th item. This item can be similar to item2 except that we directly assign `AlignSelf` property to `FlexContainer.Alignment.AlignFlexEnd`. It will show to user, that this line is not necessery thanks to inheritance.

    To add space around an item:

    ```cs
    /// Create the flex container
    FlexContainer flexContainer = new FlexContainer();
    Window.Instance.GetDefaultLayer().Add(flexContainer);

    /// Create a flex item
    View item = new View();

    /// Add a margin around the item
    item.FlexMargin = new Vector4(10.0f, 10.0f, 10.0f, 10.0f);
    item.Size2D = new Size2D(200, 200);
    item.BackgroundColor = Color.Red;
    flexContainer.Add(item);
    ```

> ## Related Information

Reviewer: Global: Kindly ensure to follow sentence case for all headings. 
@mijin85cho what's your opinion? 

```suggestion 
## Related information

Reviewer: @mijin85cho thanks a lot!

> - Example

Reviewer: We dont keep **example** in  Related Information section, kindly verify the location of: 
- Example
  - https://github.com/Samsung/Tizen-CSharp-Samples/blob/master/TV/FlexContainerSample

    To add space around an item:

    ```csharp
    /// Create the flex container
    FlexContainer flexContainer = new FlexContainer();
    Window.Instance.GetDefaultLayer().Add(flexContainer);

    /// Create a flex item
    View item = new View();

    /// Add a margin around the item
    item.FlexMargin = new Vector4(10.0f, 10.0f, 10.0f, 10.0f);
    item.Size2D = new Size2D(200, 200);
    item.BackgroundColor = Color.Red;
    flexContainer.Add(item);
    ```

## Related Information

> - Dependencies

Reviewer: @mijin85cho thanks for giving me clarity. I will note this for future.

>   -   Tizen 5.5 and Higher

Author: Since note about deprecated FlexContainer class is added I changed  dependency to Tizen 5.5 for this reason. It looks like I misunderstood  purpose of this dependency. I will restore it to Tizen 4.0 and Higher
